dna_analyse/DNA_Analyse.py

```diff
@@ -12,43 +18,81 @@
 
 def hole_dna_sequenz():
     """Liefere eine vom Nutzer erfragte und validierte DNA-Sequenz.
-
     Um gültig zu sein, darf die Sequenz nur aus den Buchstaben
     A, G, T, C, a, g, t, c bestehen.
     """
 
-    # Hinweis: Verwende eine while-Schleife, innerhalb derer Du eine
-    # Sequenz vom Nutzer erfragst und die Eingabe validierst.
-    # Nur wenn die Validierung klappt, brichst Du die Schleife ab
-    # und lieferst die Sequenz zurück.
+    # mithilfe von while-Schleife die Sequenz abfragen 
+    
+    x = 0
+    
     while True:
+        global sequence
         sequence = input('Gib eine DNA-Sequenz ein: ')
+       
+        # Überprüfen, ob alle eingegebenen Buchstaben Basen sind. 
+        # Wenn ja, müssen die Basen gezählt bzw. erst mal sortiert werden 
+        #  -> Listen 
+        # Wenn nicht: Fehlermeldung ausgeben und loop abbrechen 
         
-        # Hier fehlt natürlich noch die Validierung.
-        # Dafür brauchst Du noch eine Schleife, in der jeder Buchstabe
-        # der Eingabe auf Gültigkeit geprüft werden muss.
-        # Für einen Anfänger ist das schon eine ordentlich harte Nuss,
-        # die aber mit etwas Anstrengung zu knacken sein sollte.
-
-        if erfolgreiche_validierung:
-            break
-        else:
-            # Dieser Teil wird ausgeführt wenn die Bedingung hinter if
-            # nicht erfüllt war.
+        try:
+            while True: 
+                sequence[x] in gueltige_nukleotide 
+            
+                if sequence[x] == gueltige_nukleotide[3]:
+                    l_g.append('G')
+                elif sequence[x] == gueltige_nukleotide[0]:
+                    l_a.append('A')
+                elif sequence[x] == gueltige_nukleotide[1]:
+                    l_t.append('T')
+                elif sequence[x] == gueltige_nukleotide[2]:
+                    l_c.append ('C')
+                    
+                # Feher kreieren, wenn Buchstabe keine Base,
+                # sodass except part benutzt wird 
+                else: 
+                    sequence[x] + 2
+         
+            
+                x = x + 1 
+                if x == len(sequence):
+                    break
```

Comment:
den inneren while loop könntest Du zweckmässig durch eine for-Schleife der Art:
```
for nucleotide in sequence:
    ...
```
ersetzen und Dir die x-Hochzählerei sparen.